docs/product-assembly_running_employee_rostering_in_wb/src/main/asciidoc/wb-employee-rostering-balance-employees-shift-rule-proc.adoc

```diff
@@ -0,0 +1,32 @@
+[id='wb-employee-rostering-balance-employees-shift-rule-proc']
+= Create the `BalanceEmployeesShiftNumber` guided rule 
+
+The `BalanceEmployeesShiftNumber` guided rule creates a soft constraint that ensures all employees are assigned a shift, or that all employees are assigned an equal number of shifts. In the employee rostering example, this constraint is created using the guided rule editor. 
```

Comment:
I think we should rephrase this to help the user better understand what the rule does. The issues I can see in the description

1. ensures all employees are assigned a shift - yes, but that's a side effect, not the main point of the rule
2. all employees are assigned an equal number of shifts - ideally yes, but that's not what the rule ensures

### How the rule works (WHEN section, condition)
For each employee it counts all his/her shift assignments. The number of shifts assigned to the employee is stored in the `$shiftCount` variable.
### How the rule works (THEN section, consequence)
Soft score is modified (penalized) with the negative value of the **square** of the shift count.

This is how the *rule* works and I'm not saying this needs to be explained. It's optional in my opinion. What's more important is to explain the *idea* of the constraint implemented by the rule. The idea is to penalize extreme shift counts by using the square as a score penalty. Example:

* 3 employees with 4, 5 and 6 shifts gives a score of -16 + -25 + -36 = -77 which is a worse total score than
* 3 employees with 5, 5 and 5 shifts, giving a total score of -25 + -25 + -25 = -75. This is a higher and therefore better soft score.

This example illustrates how the score formula, implemented by the rule, gives the solver an incentive to distribute shift assignments in a balanced way. If this were the only constraint, all employees would end up being assigned to equal number of shifts. Since, in a real world data set, this probably not possible due to other constraint, I would avoid stating that the rule "makes sure all employees are assigned an equal number of shifts". "The rule ensures that shifts are assigned in a more fair, balanced way" sounds more accurate.

@gsheldon the exact wording is up to you. I'm trying some extra details to help you understand how it works.